x/mongo/driver/topology/errors.go

```diff
@@ -15,13 +19,43 @@ type ConnectionError struct {
 
 // Error implements the error interface.
 func (e ConnectionError) Error() string {
+	message := e.message
+	if e.init == true {
+		message += "error occured during connection handshake"
+	}
+	if e.Wrapped != nil && message != "" {
+		return fmt.Sprintf("connection(%s) %s: %s", e.ConnectionID, message, e.Wrapped.Error())
+	}
 	if e.Wrapped != nil {
-		return fmt.Sprintf("connection(%s) %s: %s", e.ConnectionID, e.message, e.Wrapped.Error())
+		return fmt.Sprintf("connection(%s) %s", e.ConnectionID, e.Wrapped.Error())
 	}
-	return fmt.Sprintf("connection(%s) %s", e.ConnectionID, e.message)
+	return fmt.Sprintf("connection(%s) %s", e.ConnectionID, message)
 }
 
 // Unwrap returns the underlying error.
 func (e ConnectionError) Unwrap() error {
 	return e.Wrapped
 }
+
+// ServerSelectionError represents a Server Selection error.
+type ServerSelectionError struct {
+	Desc    description.Topology
+	Wrapped error
+	message string
```

Comment:
Is the `message` field ever used?

Reply:
the `message` field is not used, I've removed it